app/repositories/application/GeneralApplicationRepoBSONToModelHelper.scala

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright 2016 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package repositories.application
+
+import model.ApplicationStatus.{apply => _, _}
+import model.ApplicationStatusOrder._
+import model.AssessmentScheduleCommands.ApplicationForAssessmentAllocation
+import model.CivilServiceExperienceType.{CivilServiceExperienceType, apply => _, _}
+import model.Commands.{Candidate, _}
+import model.InternshipType.{InternshipType, apply => _}
+import model.OnlineTestCommands.TestResult
+import model.SchemeType._
+import model.command.ProgressResponse
+import model.persisted.{ApplicationForNotification, Phase1TestProfile}
+import model.report._
+import model.{CivilServiceExperienceType, InternshipType}
+import org.joda.time.{DateTime, LocalDate}
+import reactivemongo.bson.{BSONDocument, _}
+import repositories._
+
+
+object GeneralApplicationRepoBSONToModelHelper extends GeneralApplicationRepoBSONToModelHelper
```

Comment:
(Thought): There seems to have been a lot of refactoring here, defining custom readers. Would it not have been possible to have used ReactiveMongo Macros to cut down on bolierplate readers?